dietpi/dietpi-software

```diff
@@ -6718,6 +6725,111 @@ _EOF_
+
+			# x86_64
+			elif (( $G_HW_ARCH == 10 )); then
+				arch='x86_64'
+
+			fi
+			
+			local fallback_url="https://github.com/navidrome/navidrome/releases/download/v0.47.5/navidrome_0.47.5_Linux_$arch.tar.gz"
+			Download_Install "$(curl -sSfL 'https://api.github.com/repos/navidrome/navidrome/releases/latest' | mawk -F\" "/\"browser_download_url\": .*\/navidrome_[0-9.]*_Linux_$arch\.tar\.gz\"/{print \$4}")" /opt/navidrome/
+
+			# Data dir
+			[[ -d '/mnt/dietpi_userdata/navidrome' ]] || G_EXEC mkdir /mnt/dietpi_userdata/navidrome
+
+			# User
+			Create_User -g dietpi -d /mnt/dietpi_userdata/navidrome navidrome
+
+			# Service: https://www.navidrome.org/docs/installation/linux/#create-configuration-file
+			cat << '_EOF_' > /mnt/dietpi_userdata/navidrome/navidrome.toml
+DataFolder = '/mnt/dietpi_userdata/navidrome/data'
+MusicFolder = '/mnt/dietpi_userdata/Music'
+ImageCacheSize = '100MiB'
+
+LogLevel = 'info'
+SessionTimeout = '24h'
+AuthRequestLimit = '5'
+AuthWindowLength = '20s'
+ScanSchedule = '@every 1h'
+EnableLogRedacting = 'true'
+Scanner.Extractor = 'taglib'
+
+CoverJpegQuality = '75'
+CoverArtPriority = 'embedded'
+_EOF_
+
+			# Service: https://www.navidrome.org/docs/installation/linux/#create-a-systemd-unit
+			cat << '_EOF_' > /etc/systemd/system/navidrome.service
+[Unit]
+Description=navidrome (DietPi)
+Wants=network-online.target
+After=network-online.target
+AssertPathExists=/mnt/dietpi_userdata/navidrome
+
+[Service]
+User=navidrome
+Type=simple
+ExecStart=/opt/navidrome/navidrome --configfile "/mnt/dietpi_userdata/navidrome/navidrome.toml"
+WorkingDirectory=/mnt/dietpi_userdata/navidrome
+TimeoutStopSec=20
+KillMode=process
+Restart=on-failure
+
+# See https://www.freedesktop.org/software/systemd/man/systemd.exec.html
+DevicePolicy=closed
+NoNewPrivileges=yes
+PrivateTmp=yes
+PrivateUsers=yes
+ProtectControlGroups=yes
+ProtectKernelModules=yes
+ProtectKernelTunables=yes
+RestrictAddressFamilies=AF_UNIX AF_INET AF_INET6
+RestrictNamespaces=yes
+RestrictRealtime=yes
+SystemCallFilter=~@clock @debug @module @mount @obsolete @reboot @setuid @swap
+ReadWritePaths=/mnt/dietpi_userdata/navidrome
+
+# You can uncomment the following line if you're not using the jukebox This
+# will prevent navidrome from accessing any real (physical) devices
+#PrivateDevices=yes
+
+# You can change the following line to `strict` instead of `full` if you don't
+# want navidrome to be able to write anything on your filesystem outside of
+# /mnt/dietpi_userdata/navidrome.
+ProtectSystem=full
+
+# You can uncomment the following line if you don't have any media in /home/*.
+# This will prevent navidrome from ever reading/writing anything there.
+#ProtectHome=true
+
+# You can customize some Navidrome config options by setting environment variables here. Ex:
+#Environment=ND_BASEURL="/navidrome"
+
+[Install]
+WantedBy=multi-user.target
+_EOF_
+
+			# Permissions
+			G_EXEC chown -R navidrome:dietpi /mnt/dietpi_userdata/navidrome /opt/navidrome
```

Comment:
Does it have an internal updater? If so, we'd need to skip the installation if `/opt/navidrome` exists already, else it shouldn't require write access to this file (and it would be prevented by `ProtectSystem=strict`, not set but suggested in the service):
```suggestion
			G_EXEC chown -R navidrome:dietpi /mnt/dietpi_userdata/navidrome
```